File: uhrwerk-common/src/main/java/io/qimia/uhrwerk/common/metastore/dependency/BulkDependencyResult.java
package io.qimia.uhrwerk.common.metastore.dependency;

import io.qimia.uhrwerk.common.model.Connection;
import io.qimia.uhrwerk.common.model.Dependency;
import io.qimia.uhrwerk.common.model.Partition;

import java.time.LocalDateTime;

public class BulkDependencyResult {
    LocalDateTime[] partitionTimestamps;
    Dependency dependency;
    Connection connection;
    Partition[] succeeded;

    public LocalDateTime[] getPartitionTimestamps() {
        return partitionTimestamps;
    }

    public void setPartitionTimestamps(LocalDateTime[] partitionTimestamps) {
        this.partitionTimestamps = partitionTimestamps;
    }

    public Dependency getDependency() {
        return dependency;
    }

    public void setDependency(Dependency dependency) {
        this.dependency = dependency;
    }

    public Connection getConnection() {
        return connection;
    }

    public void setConnection(Connection connection) {
        this.connection = connection;
    }

    public Partition[] getSucceeded() {
        return succeeded;
    }

    public void setSucceeded(Partition[] succeeded) {
        this.succeeded = succeeded;
    }
}

File: uhrwerk-common/src/main/scala/io/qimia/uhrwerk/common/framemanager/BulkDependencyResult.scala
package io.qimia.uhrwerk.common.framemanager

import io.qimia.uhrwerk.common.metastore.model.{ConnectionModel, DependencyModel, Partition}

import java.time.LocalDateTime

case class BulkDependencyResult(
                                 partitionTimestamps: Array[LocalDateTime],
                                 dependency: DependencyModel,
                                 connection: ConnectionModel,
                                 succeeded: Array[Partition]
                               )